backend/src/main/java/com/zeros/backend/service/Route.java

```diff
@@ -0,0 +1,64 @@
+package com.zeros.backend.service;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.util.Objects;
+
+public class Route  {
+
+	private final BigDecimal distance;
+	private final Duration duration ;
+
+	private Route (newBuilder  builder) {
+		this.distance = builder.distance;
+		this.duration = builder.duration;
+	}
+
+	public BigDecimal getDistance() {
+		return distance;
+	}
+
+	public Duration getDuration() {
+		return duration;
+	}
+	
+	@Override
+    public boolean equals(Object o) {
+		if (o == this) return true;
+		if (!(o instanceof Route)) {
+			return false;
+		}
+		Route route = (Route) o;
+		return Objects.equals(distance, route.distance) &&
+                Objects.equals(duration, route.duration);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(distance, duration);
+    }
+
+	public static class newBuilder{
+		private BigDecimal distance;
```

Comment:
I don't feel that we need a builder here. The construction of object is not complicated

Reply:
I always prefer using builders :)

Reply:
Builder are useful when a lot of parameter should be set when constructing the object, however in Route we only have distance and duration we can use constructor directly. I think that @zeroual is right on this one